File: 0050-powx-n/0050-powx-n.java
class Solution {
    public double myPow(double x, int n) {
        long binform=n;
        if(n<0)
        {
            x=1/x;
            binform=-binform;
        }
        double ans=1;
        while(binform>0)
        {
            if(binform%2==1)
            {
                ans*=x;
            }
            x*=x;
            binform/=2;
        }
        return ans;
    }
}

File: 0050-powx-n/0050-powx-n.py
class Solution(object):
    def myPow(self, x, n):
        """
        :type x: float
        :type n: int
        :rtype: float
        """
        ans=1
        binform=int(n)
        if n<0:
            x=1/x
            binform=-binform
        while binform>0:
            if binform%2==1:
                ans*=x
            x*=x
            binform/=2
        return ans